### PR TITLE
fix modal select overflow position bug

### PR DESCRIPTION
**Pull request by thescubageek**
* Fixes materialize select overflow bug in modals where dropdown was being mostly obscured by modal `overflow: hidden` -- the following logic is applied:
  * use 300ms timeout before adjusting select dropdown (300ms is the default animation speed for dropdown to slide out in materialize -- less timeout than this results in incorrect height calculations)
  * if select dropdown fits without overflow, leave it alone
  * if select dropdown overflows but fits inside `.modal-content` height, then slide up until the bottom of the dropdown == bottom of `.modal-content`
  * if select dropdown overflows and is larger than `.modal-content` height, then slide up to fill entire `.modal-content`

I'm not a fan of having to use `this.$().parents('.modal-content');` to access the modal content outside of the component, but there's not a clean way to get this calcs to work without doing this hack (at least not that I've solved thus far... took me forever to figure out this hack -- might need to test against mobile [if we support mobile for CMS] since `mouseup` event probably won't work [`mouseup` was the only event that was reliable])

PT: https://www.pivotaltracker.com/story/show/131687435

### CODE REVIEW

**Comment by thescubageek** [comment]
@jcstringer @lance-hardy Check out this hack to get select modals working with respect to modal heights and overflow

**Comment by lance-hardy** [comment]
I can imagine a mod for materialize itself would be a snappy contribution to that project

**Comment by jcstringer** [comment]
@thescubageek nice

File: app/components/widget-edit-html.js [hunk 1]
```diff
@@ -44,6 +44,33 @@ export default Ember.Component.extend(Widgetable, {
     }
   },
 
+  setupSelects(){
+    this.$('select').material_select();
+    this.$('.select-wrapper').on('mouseup', (e)=>{
```
Comment (g5codyswartz) [inline app/components/widget-edit-html.js]: mousedown event may be more responsive and less prone to issues when the select pops up and the mouseup event can no longer trigger (if you click and hold on the select box the event will never trigger)

**Pull request merged**